controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -49,13 +53,187 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
+
+	// init revisions
+	if nsTmplTier.Status.Revisions == nil {
+		nsTmplTier.Status.Revisions = map[string]string{}
+	}
+	// check for TierTemplates and TierTemplateRevisions associated with the NSTemplateTier
+	ttrCreated := false
+	for _, tierTemplateRef := range refs {
+		// get the TierTemplate
+		var tierTemplate toolchainv1alpha1.TierTemplate
+		if err := r.Client.Get(ctx, types.NamespacedName{Namespace: nsTmplTier.GetNamespace(), Name: tierTemplateRef}, &tierTemplate); err != nil {
+			// something went wrong or we haven't found the TierTemplate
+			return false, err
+		}
+
+		// check if there is TTR associated with this TierTemplate
+		ttrCreatedLatest, ttrName, err := r.ensureTTRforTemplate(ctx, nsTmplTier, &tierTemplate)
+		ttrCreated = ttrCreated || ttrCreatedLatest
+		if err != nil {
+			return false, err
+		}
+		nsTmplTier.Status.Revisions[tierTemplate.GetName()] = ttrName
+	}
+	// TODO handle removal of TierTemplate from NSTemplateTier
+	// scenario:
+	// 		a. TierTemplate is removed/replaced from NSTemplateTier.Spec
+	//		b. NSTemplateTier.Status.Revisions must be cleaned up
+	// Thus here we should iterate over the Status.Revisions field
+	// and check if there is any reference to a TierTemplate that is not in the Spec anymore
+	if ttrCreated {
+		// we need to update the status.revisions with the new ttrs
+		logger.Info("ttr created updating status")
+		if err := r.Client.Status().Update(ctx, nsTmplTier); err != nil {
+			return ttrCreated, err
+		}
+	}
+
+	return ttrCreated, nil
+}
+
+func (r *Reconciler) ensureTTRforTemplate(ctx context.Context, nsTmplTier *toolchainv1alpha1.NSTemplateTier, tierTemplate *toolchainv1alpha1.TierTemplate) (bool, string, error) {
+	logger := log.FromContext(ctx)
+	// tierTemplate doesn't support TTRs
+	// we set TierTemplate as revisions
+	// TODO this step will be removed once we convert all TierTemplates to TTRs
+	if tierTemplate.Spec.TemplateObjects == nil {
+		_, ok := nsTmplTier.Status.Revisions[tierTemplate.GetName()]
+		if !ok {
+			return true, tierTemplate.GetName(), nil
+		}
+		// nothing to update
+		return false, "", nil
+	}
+
+	if tierTemplateRevisionName, found := nsTmplTier.Status.Revisions[tierTemplate.GetName()]; found {
+		logger.Info("TTR set in the status.revisions for tiertemplate", "tierTemplate.Name", tierTemplate.GetName(), "ttr.Name", tierTemplateRevisionName)
+		var tierTemplateRevision toolchainv1alpha1.TierTemplateRevision
+		if err := r.Client.Get(ctx, types.NamespacedName{Namespace: nsTmplTier.GetNamespace(), Name: tierTemplateRevisionName}, &tierTemplateRevision); err != nil {
+			if errors.IsNotFound(err) {
+				// no tierTemplateRevision CR was found,
+				logger.Info("TTR CR not found", "tierTemplateRevision.Name", tierTemplateRevisionName)
+				// let's create one
+				ttrName, err := r.createNewTierTemplateRevision(ctx, nsTmplTier, tierTemplate)
+				if err != nil {
+					return false, "", err
+				}
+				// new ttr created
+				return true, ttrName, nil
```

Comment:
could be simplified to:
```suggestion
				return true, ttrName, err
```

Reply:
good catch, thanks for the suggestion 👍